VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -158,6 +159,25 @@ QList<QPair<Item*, Item*>> ViewItem::arrowsForLayer(QString layer)
 	return arrows_[layer];
 }
 
+void ViewItem::determineChildren()
+{
+	Super::determineChildren();
+	//Update all the info nodes in the view
+	for (auto node : allNodes())
+	{
+		auto asViewItemNode = DCast<ViewItemNode>(node);
+		Q_ASSERT(asViewItemNode);
+		if (auto info = DCast<InfoNode>(asViewItemNode->reference()))
+		{
+			//Update only if we don't already do so by default
+			if (auto vis = findVisualizationOf(info))
+				//determineChildren is protected in Item, but not in DeclarativeItemBase
+				if (vis->needsUpdate() == Item::NoUpdate)
+					DCast<DeclarativeItemBase>(vis)->determineChildren();
```

Comment:
The reason I put the distinction is so that on an automatic update, not necessarily everything will be updated (specifically, updates which might be computationally expensive to perform every cycle). I am not entirely sure what you mean by distinguishing this in the node itself - the updateInfo(false/true) already does that, no?

The first proposal certainly seems easier than my solution though.